src/deposit_monitor.rs

```diff
@@ -63,11 +84,49 @@ impl DepositMonitor {
 
         let bitcoin_client = self.context.bitcoin_client();
 
-        // TODO: cache results
-        let block_hash = bitcoin_client.get_block_hash(utxo.block_height)?;
-
-        // TODO: cache results
-        let tx_hex = bitcoin_client.get_raw_transaction_hex(&utxo.txid, &block_hash)?;
+        let cached_hash = self.hashes_cache.get(&utxo.block_height);
+        let block_hash = match cached_hash {
+            Some(hash) => {
+                tracing::debug!(
+                    "Using cached block hash for utxo at height {}: {}",
+                    utxo.block_height,
+                    hash
+                );
+                *hash
+            }
+            None => {
+                tracing::debug!(
+                    "Fetching block hash for utxo at height {}",
+                    utxo.block_height
+                );
+                let hash = bitcoin_client.get_block_hash(utxo.block_height)?;
+                self.hashes_cache.put(utxo.block_height, hash);
+                hash
+            }
+        };
+
+        let cached_tx_hex = self.tx_hex_cache.get(&(utxo.txid, block_hash));
+        let tx_hex = match cached_tx_hex {
+            Some(hex) => {
+                tracing::debug!(
+                    "Using cached transaction hex for txid {} at block {}",
+                    utxo.txid,
+                    block_hash
+                );
+                hex.clone()
+            }
+            None => {
+                tracing::debug!(
+                    "Fetching transaction hex for txid {} at block {}",
+                    utxo.txid,
+                    block_hash
+                );
+                let tx_hex = bitcoin_client.get_raw_transaction_hex(&utxo.txid, &block_hash)?;
+                self.tx_hex_cache
+                    .put((utxo.txid, block_hash), tx_hex.clone());
+                tx_hex
+            }
+        };
```

Comment:
I don't think we need logs for this, and we could use ` try_get_or_insert` to slim this down

Reply:
whoops, accidentally left debug logs, [cb6d30f](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/cb6d30fceaa37cabc39323849f699f7fb686c9f8), [d68606b](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/d68606bc7b2fabd4bcd695cc9eb90f1c9bf24024)